tests/httpfake.h

```diff
@@ -26,10 +27,13 @@ class HttpFake : public HttpInterface {
       : test_dir(test_dir_in), flavor_(std::move(flavor)), meta_dir(meta_dir_in) {
     if (meta_dir.empty()) {
       meta_dir = temp_meta_dir.Path();
-      Utils::copyDir("tests/test_data/repo/repo/image", meta_dir / "repo");
-      Utils::copyDir("tests/test_data/repo/repo/director", meta_dir / "director");
-      if (boost::filesystem::is_directory("tests/test_data/repo/campaigner")) {
-        Utils::copyDir("tests/test_data/repo/campaigner", meta_dir / "campaigner");
+
+      boost::filesystem::path gen_dir = meta_dir / "meta_fake";
+      MetaFake meta(gen_dir);
```

Comment:
Unfortunately, this is making the test times to explode and even times out in CI.

My suggestion to use the native lib actually made it worse because it then runs under valgrind in the test suite, which slows it down even more.

But in any case, `MetaFake()` should be moved somewhere outside this constructor so that it can be shared between test cases. The output directory can then just be passed as an argument to `HttpFake::HttpFake()`.